t/22-svirt.t

```diff
@@ -1034,4 +1034,136 @@ subtest 'get_wait_still_screen_on_here_doc_input' => sub {
+            # run shell script in local dir.
+            my $out;
+            $out = qx($cmd);
+            $out;
+    });
+
+    $out2 = $svirt->provide_image_vmware_in_ds($input_file_xz, $vmware_openqa_datastore, backingfile => 1);
+    like $last_ssh_commands[0], qr/\/hdd\/$input_file_xz\s/, 'Checking image management script with xz- 5a';
+    like $out2, qr/$vmware_openqa_datastore\/$input_file/, 'Checking image management output with xz- 5b';
+
+    path($vmware_openqa_datastore, $input_file)->remove;
+
+    $out2 = $svirt->provide_image_vmware_in_ds($input_file, $vmware_openqa_datastore, backingfile => 1);
+    like $last_ssh_commands[1], qr/\/hdd\/$input_file\s/, 'Checking image management script without xz- 6a';
+    like $out2, qr/$vmware_openqa_datastore\/$input_file/, 'Checking image management output without xz- 6b';
+
+    path($vmware_openqa_datastore, $input_file)->remove;
+
+    $out2 = $svirt->provide_image_vmware_in_ds($input_file, $vmware_openqa_datastore, backingfile => 0);
+    like $last_ssh_commands[2], qr/\/iso\/$input_file\s/, 'Checking image management script without xz- 7a';
+    like $out2, qr/$vmware_openqa_datastore\/$input_file/, 'Checking image management output without xz- 7b';
+
+    path($vmware_openqa_datastore, $input_file)->remove;
+
+    $out2 = $svirt->provide_image_vmware_in_ds($input_file_xz, $vmware_openqa_datastore);
+    like $last_ssh_commands[3], qr/\/iso\/$input_file_xz\s/, 'Checking image management script with xz- 8a';
+    like $out2, qr/$vmware_openqa_datastore\/$input_file/, 'Checking image management output with xz- 8b';
+
+    # $out2 = $svirt->provide_image_vmware_in_ds($input_file, $vmware_openqa_datastore);
+    # like $last_ssh_commands[4], qr/\/iso\/$input_file_xz\s/, 'Checking image management script with xz- 9a';
+    # like $out2, qr/$vmware_openqa_datastore\/$input_file/, 'Checking image management output with xz- 9b';
+
+};
+
+
+subtest 'Routine consoles::sshVirtsh::provide_image_vmware_in_ds' => sub {
+    set_var(VMWARE_NFS_DATASTORE_DEBUG => 1);
+
+    my $my_test_basedir = tempdir($dir . '/tb_XXXX');    # base subdir of testing, default /vmfs/volumes.
+    set_var(VIRSH_OPENQA_BASEDIR => $my_test_basedir);
+
+    # orig. like: /vmfs/volumes/openqa/hdd/img.vmdk.xz
+    my $nfs_ds = 'openqa100';
+    set_var(VMWARE_NFS_DATASTORE => $nfs_ds);
+
+    my $my_test_dir = path($my_test_basedir, $nfs_ds);
+    my $my_test_dir_hdd = path($my_test_dir, 'hdd');
+    my $my_test_dir_iso = path($my_test_dir, 'iso');
+
+    # dest: /vmfs/volumes/Datastore1/openQA/img.vmdk.xz
+    my $ds = 'datastore100';
+    set_var(VMWARE_DATASTORE => $ds);
+    my $vmware_openqa_datastore = path($my_test_basedir, $ds, 'openQA');
+
+    my $input_file = path("vmware-mock100-image.vmdk");
+    my $input_file_xz = path($input_file . '.xz');
+
+    my $svirt = consoles::sshVirtsh->new('svirt');
+    my $console_mock = Test::MockModule->new('consoles::sshVirtsh');
+    my @last_ssh_commands = ();
```

Comment:
this could be extracted in a function to avoid duplication.

Reply:
The 2 tests are different, but I will try to refactor the test in a basic main test, with higer scope settings in the heading,  nesting the other sub tests inside, to use those common vaules.